### PR TITLE
Open edX Dogwood devstack and fullstack ARM templates. Originally cre…

### PR DESCRIPTION
**Pull request by chenriksson**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Description of the change
Open edX Dogwood devstack and fullstack ARM templates. Originally created by Appsembler for Microsoft.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@chenriksson__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (chenriks). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by chenriksson** [comment]
Per Travis CI: If you require sudo, add 'sudo: required' to your .travis.yml

These templates do require sudo. To fix, am I supposed to add a .travis.yml file to the subdirectory for these templates?

**Comment by gatneil** [comment]
custom script extensions get run as root without you having to use the sudo command. That logging from the Travis CI isn't referring to the custom script extension :)

File: openedx-devstack-ubuntu/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,262 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
```
Comment (singhkay) [inline openedx-devstack-ubuntu/azuredeploy.json]: This should be auto-generated. See #944

File: openedx-devstack-ubuntu/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        "description": "The name of the new storage account created to store the VMs disks. Name must be between 3 and 24 characters and use lowercase alphanumeric only."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Administrator password. Username is 'vagrant'."
+      }
+    },
+    "dnsNameForPublicIP": {
```
Comment (singhkay) [inline openedx-devstack-ubuntu/azuredeploy.json]: See #966

File: openedx-devstack-ubuntu/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+      }
+    }
+  },
+  "variables": {
+    "apiVersion": "2015-06-15",
+    "openedxVersion": "named-release/dogwood.1",
+    "scriptDownloadUri": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/openedx-devstack-ubuntu/master/",
+    "installScript": "install-openedx.sh",
+    "installCommand": "[concat('sudo bash -c ''nohup ./', variables('installScript'), ' ', variables('openedxVersion'), ' </dev/null &>/var/log/azure/openedx-install.log &''')]",
+    "adminUsername": "vagrant",
```
Comment (singhkay) [inline openedx-devstack-ubuntu/azuredeploy.json]: This should be a parameter for security

File: openedx-devstack-ubuntu/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    "vmName": "jumpbox2",
+    "osImagePublisher": "Canonical",
+    "osImageOffer": "UbuntuServer",
+    "osImageSKU" : "12.04.5-LTS",
```
Comment (singhkay) [inline openedx-devstack-ubuntu/azuredeploy.json]: Any reason to use 12 instead of 14 LTS?

File: openedx-devstack-ubuntu/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat(variables('vmName'), '/installscript')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', variables('vmName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.4",
```
Comment (singhkay) [inline openedx-devstack-ubuntu/azuredeploy.json]: please add autoupgrademinorversion

File: openedx-devstack-ubuntu/azuredeploy.json (hunk 4 above)
Comment (chenriksson) [inline openedx-devstack-ubuntu/azuredeploy.json]: Open edX installation docs specifically call out 12: https://openedx.atlassian.net/wiki/display/OpenOPS/Open+edX+Installation+Options

Saw in forums that some users are on 14, but I'm hoping to do this in a future commit once I've done more testing. I do realize Azure support for 12 ends on 4/17:
https://azure.microsoft.com/en-us/marketplace/partners/canonical/ubuntuserver1204lts/

**Comment by chenriksson** [comment]
Thanks for your help! Looks like the CI validation is passing, and it's just failing with 404 when trying to download the raw.githubusercontent script which isn't merged yet. Please let me know if there's anything I need to do.

**Comment by gatneil** [comment]
It looks like the first error of 8 is the 404, but there also seem to be other errors (mismatched parameters, incorrectly referenced vnet, etc.)


-Neil

[&#X1f60a][&#X1f609][&#X02639][&#X1f61d][&#X1f60e]

________________________________

**Comment by chenriksson** [comment]
Thanks, user error - forgot to sync my fork.

**Comment by chenriksson** [comment]
Ping - I think the only remaining failures are the 2 404s for the 2 templates. Anything else I should do?

File: openedx-devstack-ubuntu/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+  "variables": {
+    "apiVersion": "2015-06-15",
+    "openedxVersion": "named-release/dogwood.1",
+    "scriptDownloadUri": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/openedx-devstack-ubuntu/master/",
+    "installScript": "install-openedx.sh",
+    "installCommand": "[concat('bash -c ''nohup ./', variables('installScript'), ' ', variables('openedxVersion'), ' </dev/null &>/var/log/azure/openedx-install.log &''')]",
+    "vmName": "jumpbox2",
+    "osImagePublisher": "Canonical",
+    "osImageOffer": "UbuntuServer",
+    "osImageSKU" : "12.04.5-LTS",
```
Comment (singhkay) [inline openedx-devstack-ubuntu/azuredeploy.json]: Why not use 14.04 LTS?

File: openedx-devstack-ubuntu/LICENSE [hunk 7]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
```
Comment (singhkay) [inline openedx-devstack-ubuntu/LICENSE]: I don't think a license per template is needed. The repo has a MIT license as well.

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**